src/monitor/group_state_machine.c

```diff
@@ -1028,13 +1082,55 @@ ProceedGroupStateForMSFailover(AutoFailoverNode *activeNode,
 
 	/*
 	 * If a failover is in progress, continue driving it.
+	 *
+	 * The currently selected node might not be marked healthy at this time
+	 * because in REPORT_LSN we shut Postgres down. We still should proceed
+	 * with the previously selected node in that case.
+	 *
+	 * Also pay attention that Postgres might be running again, as reported by
+	 * the pg_autoctl node-active protocol, while our health checks have not
+	 * picked-up on that fact yet. The IsHealthy() check requires both checks
+	 * to be all-green, finding the candidate is okay with only one of those
+	 * being good: we really need to avoid having two candidates at the same
+	 * time.
```

Comment:
I believe the changes that this comment addresses have been (re)moved again. I don't see anything related to `isRunning` and the `REPORT_LSN` stuff is addressed below in a second comment already.

Reply:
Good point, yeah, this part of the code received several changes following some QA sessions and testing. Fixed now.